icu4c/source/test/intltest/unitstest.cpp

```diff
@@ -37,54 +44,73 @@ void UnitsTest::runIndexedTest(int32_t index, UBool exec, const char *&name, cha
     TESTCASE_AUTO_END;
 }
 
-// Just for testing quick conversion ability.
-double testConvert(UnicodeString source, UnicodeString target, double input) {
-    if (source == u"meter" && target == u"foot" && input == 1.0)
-        return 3.28084;
+// Assert if two dec numbers are equal.
+void UnitsTest::assertConversion(StringPiece message, StringPiece source, StringPiece target,
+                                 double inputValue, double expectedValue) {
+    UErrorCode status;
+
+    MeasureUnit sourceUnit = MeasureUnit::forIdentifier(source, status);
+    MeasureUnit targetUnit = MeasureUnit::forIdentifier(target, status);
+
+    // assertSuccess("test Convert: Units Creation", status);
 
-    if ( source == u"kilometer" && target == u"foot" && input == 1.0)
-        return 328.084;
+    UnitConverter converter(sourceUnit, targetUnit, status);
+    // assertSuccess("test Convert: Unit Converter Creation", status);
 
-    return -1;
+    decNumber actualConversionResult = converter.convert(inputValue, status);
+    // assertSuccess("test Convert: Converter Operation has been done!", status);
+
+    number::impl::DecNum decNum;
+    decNum.setTo(expectedValue, status);
+    decNumber expectedValueDecNumber = *(decNum.getRawDecNumber());
```

Comment:
Avoid using the `decNumber` type, *especially* not by value, unless you know what you are doing.  It is a low-level C type that assumes it has room to store digits on the stack, but if you don't, it will just clobber surrounding variables.  Not good; that's why I made the nicer DecNum C++ wrapper for it.

For testing equality, consider converting the DecNum to a string and comparing it by string value.